src/resolve/source.rs

```diff
@@ -198,11 +257,29 @@ fn to_include(
     }
 }
 
-fn to_path<P: AsRef<Path>>(path: &str, workdir: P) -> io::Result<PathBuf> {
-    let path = Path::new(&path);
-    let old_workdir = env::current_dir()?;
-    env::set_current_dir(workdir)?;
-    let path = path.canonicalize()?;
-    env::set_current_dir(old_workdir)?;
-    Ok(path)
+fn to_path<P: AsRef<Path>>(url: &Url, workdir: P) -> std::result::Result<PathBuf, PathError> {
+    let mut full_path = workdir.as_ref().to_path_buf();
+
+    url
+        .path_segments()
+        .ok_or(PathError::NoPath)?
+        .try_for_each(|segment| {
+            let mut components = Path::new(segment)
+                .components();
```

Comment:
Is this really needed? `Url::path_segments` should already split at `/`, which should be enough.

Reply:
`path_segments` is split at `/` but could contain other OS specific separators or markers such as on Windows a literal `\` , a file drive component `C:`, ... Basically, I wanted to make this as safe as possible and avoid any ambiguity.